lib/fast_jsonapi/serialization_core.rb

```diff
@@ -130,6 +128,22 @@ def get_included_records(record, includes_list, known_included_objects)
           end
         end
       end
+
+      def fetch_associated_object(record, relationship)
+        return relationship[:object_block].call(record) unless relationship[:object_block].nil?
+        record.send(relationship[:object_method_name])
+      end
+
+      def fetch_id(record, relationship)
+        unless relationship[:object_block].nil?
+          object = relationship[:object_block].call(record)
+
+          return object.map(&:id) if object.respond_to? :map
+          return object.id
```

Comment:
@shishirmk is this designed to work that returns `nil`? If I return `nil` in the block, this throws a runtime error.

Reply:
Thats something we would have to fix.